src/D2L.CodeStyle.Analyzers/Language/AttributeAliasesAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+			context.RegisterCompilationStartAction( CompilationStart );
+		}
+
+		private void CompilationStart( CompilationStartAnalysisContext context ) {
+
+			context.RegisterSyntaxNodeAction(
+					c => AnalyzeAttribute( c, (AttributeSyntax)c.Node ),
+					SyntaxKind.Attribute
+				);
+		}
+
+		private void AnalyzeAttribute(
+				SyntaxNodeAnalysisContext context,
+				AttributeSyntax attribute
+			) {
+
+			// if it's not an identifier, then it's qualified and not aliasing the attribute name
+			if( attribute.Name is not IdentifierNameSyntax attributeName  ) {
+				return;
+			}
+
+			IEnumerable<IdentifierNameSyntax> usingAliases = GetUsingAliases( attribute, context.CancellationToken  );
+			foreach( IdentifierNameSyntax usingAlias in usingAliases ) {
+
+				if( !IsEquivalentToUsingAlias( attributeName, usingAlias ) ) {
+					continue;
+				}
+
+				Diagnostic d = Diagnostic.Create(
+						Diagnostics.AliasingAttributeNamesNotSupported,
+						attribute.GetLocation()
+					);
+
+				context.ReportDiagnostic( d );
+				return;
+			}
+		}
+
+		private static bool IsEquivalentToUsingAlias(
+				IdentifierNameSyntax attributeIdentifier,
+				IdentifierNameSyntax usingAliasIdentifier
+			) {
+
+			string attributeName = attributeIdentifier.ToString();
+			string usingAlias = usingAliasIdentifier.ToString();
+
+			if( attributeName.Equals( usingAlias, StringComparison.Ordinal ) ) {
+				return true;
+			}
+
+			const string attributeSuffix = "Attribute";
+			if( !usingAlias.EndsWith( attributeSuffix, StringComparison.Ordinal ) ) {
+				return false;
+			}
+
+			int shortUsingAliasLength = usingAlias.Length - attributeSuffix.Length;
+			if( shortUsingAliasLength != attributeName.Length ) {
+				return false;
+			}
+
+			bool equivalent = string.CompareOrdinal( attributeName, 0, usingAlias, 0, shortUsingAliasLength ) == 0;
+			return equivalent;
+		}
+
+		private IEnumerable<IdentifierNameSyntax> GetUsingAliases(
+				AttributeSyntax attribute,
+				CancellationToken cancellationToken
+			) {
+
+			IEnumerable<UsingDirectiveSyntax> rootUsingDirectives = attribute
+				.SyntaxTree
+				.GetCompilationUnitRoot( cancellationToken )
+				.Usings;
+
+			IEnumerable<UsingDirectiveSyntax> namespacedUsingDirectives = attribute
+				.Ancestors()
+				.OfType<BaseNamespaceDeclarationSyntax>()
+				.SelectMany( @namespace => @namespace.Usings );
+
+			IEnumerable<UsingDirectiveSyntax> allUsingDirectives = Enumerable
+				.Concat( rootUsingDirectives, namespacedUsingDirectives );
+
+			foreach( UsingDirectiveSyntax usingDirective in allUsingDirectives ) {
+
+				NameEqualsSyntax? alias = usingDirective.Alias;
+				if( alias == null ) {
+					continue;
+				}
+
+				// ignore aliases that just import the class from the namespace
+				SimpleNameSyntax unqualifiedUsingName = usingDirective.Name.GetUnqualifiedName();
+				if( StringComparer.Ordinal.Equals( alias.Name.ToString(), unqualifiedUsingName.ToString() ) ) {
+					continue;
+				}

```

Comment:
Extension method on `NameSyntax`? It seems like this would need to get re-used a bunch.

Reply:
Will need to play to feel things out.  Will depend on the next PR too.  Whether we just want to check both, or keep it simple with one.